src/core/security/auth.c

```diff
@@ -202,6 +189,7 @@ static void call_op(grpc_call_element *elem, grpc_call_element *from_elem,
                          "Invalid host %s set in :authority metadata.",
                          call_host);
             bubbleup_error(elem, error_msg);
+            grpc_metadata_batch_destroy(&calld->op.data.metadata);
```

Comment:
remove that call to destroy if you decide to put it in bubbleup_error (which I think is the right thing to do).